code/modules/projectiles/projectile.dm

```diff
@@ -57,6 +57,16 @@
 	var/forcedodge = 0 //to pass through everything
 	var/dismemberment = 0 //The higher the number, the greater the bonus to dismembering. 0 will not dismember at all.
 	var/impact_effect_type //what type of impact effect to show when hitting something
+
+	///Does the projectile knock back mobs on hit?
+	var/knockback = FALSE
+	///Does the projectile throw back objects with move resist less than INFINITE on hit?
+	var/breaching = FALSE
```

Comment:
I feel like these two can be combined into one var using bit flags.
so something like this:
```dm
#define PROJECTILE_KNOCKBACK_MOB (1<<0)
#define PROJECTILE_KNOCKBACK_OBJ (1<<1)
...

knockback = PROJECTILE_KNOCKBACK_MOB | PROJECTILE_KNOCKBACK_OBJ
...
if(((knockback & PROJECTILE_KNOCKBACK_MOB) && isliving(target)) || ((knockback & PROJECTILE_KNOCKBACK_OBJ) && isobj(target)))
  do knockback stuff...
```

The defines would be in one of the `__DEFINE` files, or a new one if none of the existing files are suitable. This way all other files can use the defines.